rust/ql/src/queries/unusedentities/UnusedValueBad.rs

```diff
@@ -0,0 +1,10 @@
+fn get_sum(values:&[i32]) -> i32 {
+	let mut sum = 0;
+	let mut average = 0; // BAD: unused value
+
+	for v in values {
+		sum += v;
+	}
+
+	return sum;
+}
```

Comment:
This is the same snippet as for the unused variable query. Wouldn't the actual example involve a reassignment to `average`?
```suggestion
fn get_average(values:&[i32]) -> i32 {
	let mut sum = 0;
	let mut average = 0; // BAD: unused value

	for v in values {
		sum += v;
	}

        average = sum / values.len();
	return average;
}
```

Reply:
It's not quite the same because one of them initializes the value to zero, the other does not.  I think something your example is clearer though.